libbeat/kibana/client.go

```diff
@@ -145,8 +145,10 @@ func NewClientWithConfig(config *ClientConfig) (*Client, error) {
 		},
 	}
 
-	if err = client.SetVersion(); err != nil {
-		return nil, fmt.Errorf("fail to get the Kibana version: %v", err)
+	if !config.IgnoreVersion {
```

Comment:
Can you remind me in which case we need this?

Reply:
basically when doing enrollment, Kibana will probably be using x-pack security, so you need user/password to access the version, we don't have it, as enrollment endpoints don't require username/password but a valid token.

This change allows to configure the client to ignore kibana version check when doing enrollment

Reply:
Got it. Chicken / egg problem here :-)